From 478aa7dac12fbbb01161a9bc78b70b8f7746fef9 Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Wed, 20 Mar 2013 06:41:25 +0100
Subject: [PATCH] 51-dracut-rescue-postinst.sh,51-dracut-rescue.install: use
 systemd-machine-id-setup

if /etc/machine-id is empty, use systemd-machine-id-setup
---
 51-dracut-rescue-postinst.sh | 5 +++++
 51-dracut-rescue.install     | 6 ++++++
 2 files changed, 11 insertions(+)

diff --git a/51-dracut-rescue-postinst.sh b/51-dracut-rescue-postinst.sh
index 8b4e6c3..e290f1a 100755
--- a/51-dracut-rescue-postinst.sh
+++ b/51-dracut-rescue-postinst.sh
@@ -8,6 +8,11 @@ KERNEL_VERSION="$1"
 KERNEL_IMAGE="$2"
 
 [[ -f /etc/os-release ]] && . /etc/os-release
+
+if [[ ! -f /etc/machine-id ]] || [[ ! -s /etc/machine-id ]]; then
+    systemd-machine-id-setup
+fi
+
 [[ -f /etc/machine-id ]] && read MACHINE_ID < /etc/machine-id
 
 [[ $MACHINE_ID ]] || exit 1
diff --git a/51-dracut-rescue.install b/51-dracut-rescue.install
index 1c0a6e1..39d01a7 100755
--- a/51-dracut-rescue.install
+++ b/51-dracut-rescue.install
@@ -31,7 +31,13 @@ dropindirs_sort()
 }
 
 [[ -f /etc/os-release ]] && . /etc/os-release
+
+if [[ ! -f /etc/machine-id ]] || [[ ! -s /etc/machine-id ]]; then
+    systemd-machine-id-setup
+fi
+
 [[ -f /etc/machine-id ]] && read MACHINE_ID < /etc/machine-id
+
 if [[ -f /etc/kernel/cmdline ]]; then
     readarray -t BOOT_OPTIONS < /etc/kernel/cmdline
 fi
